Sources/ProjectSpec/TargetReference.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+import JSONUtilities
+
+public struct TargetReference: Hashable {
+    public var name: String
+    public var location: Location
+
+    public enum Location: Hashable {
+        case local
+        case project(String)
+    }
+
+    public init(name: String, location: Location) {
+        self.name = name
+        self.location = location
+    }
+}
+
+extension TargetReference {
+    public init(string: String) throws {
```

Comment:
We can probably drop the `string` param label
```suggestion
    public init(_ string: String) throws {
```